File: app/frontend/src/api/analyze.ts
export interface Finding {
  check_id: string
  severity: string
  file_path: string
  line: number | null
  title: string
  explanation: string
  evidence: string
  suggestion: string
}

export interface AnalysisSummary {
  total: number
  high: number
  medium: number
  low: number
}

export interface AnalyzeResponse {
  repository_path: string
  summary: AnalysisSummary
  findings: Finding[]
}

export async function analyzeRepository(
  repositoryPath: string,
): Promise<AnalyzeResponse> {
  const response = await fetch('http://127.0.0.1:8000/api/analyze', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      repository_path: repositoryPath,
    }),
  })

  if (!response.ok) {
    throw new Error(`Analysis failed with status ${response.status}`)
  }

  return response.json()
}

File: app/backend/models/analyze.py
"""Pydantic models for POST /api/analyze."""
from typing import Optional
from pydantic import BaseModel


class AnalyzeRequest(BaseModel):
    """Request body for POST /api/analyze."""

    repository_path: str


class FindingOut(BaseModel):
    """Serialised representation of a single analyzer Finding."""

    check_id: str
    severity: str
    file_path: str
    line: Optional[int]
    title: str
    explanation: str
    evidence: str
    suggestion: str


class AnalysisSummary(BaseModel):
    """Aggregate counts across all findings."""

    total: int
    high: int
    medium: int
    low: int


class AnalyzeResponse(BaseModel):
    """Response body for POST /api/analyze."""

    repository_path: str
    summary: AnalysisSummary
    findings: list[FindingOut]
